data.tf

```diff
@@ -0,0 +1,26 @@
+data "aws_caller_identity" "current" {}
+
+data "aws_partition" "current" {}
+
+data "aws_iam_policy_document" "combined" {
+  source_policy_documents = compact([data.aws_iam_policy_document.default.json, var.bucket_policy])
+}
+
+data "aws_iam_policy_document" "default" {
+  statement {
```

Comment:
@patrickmukumbu why this policy here by default?

Reply:
This was a work around for using `count` condition to create the policy. When part of the policy is dynamically created, in this case the random name....it brought an error that `error.....count attributes cannot be determined until apply` @hugoalmeida264